### PR TITLE
Replace use of `Result.env` in profiling code

### LINKED ISSUE #801: running profile without commit results in obscure  AttributeError: 'Results' object has no attribute 'env'
```shell
$> asv profile  api.testds.time_create_test_dataset2x2      
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad/venvs/dev/bin/asv", line 11, in <module>
    load_entry_point('asv', 'console_scripts', 'asv')()
  File "/home/yoh/proj/misc/asv/asv/main.py", line 38, in main
    result = args.func(args)
  File "/home/yoh/proj/misc/asv/asv/commands/__init__.py", line 49, in run_from_args
    return cls.run_from_conf_args(conf, args)
  File "/home/yoh/proj/misc/asv/asv/commands/profiling.py", line 94, in run_from_conf_args
    **kwargs)
  File "/home/yoh/proj/misc/asv/asv/commands/profiling.py", line 148, in run
    for env in environments)
  File "/home/yoh/proj/misc/asv/asv/commands/profiling.py", line 148, in <genexpr>
    for env in environments)
AttributeError: 'Results' object has no attribute 'env'
```
and runs ok if I give HEAD
```shell
$> asv profile api.testds.time_create_test_dataset2x2 HEAD
· Discovering benchmarks
·· Uninstalling from virtualenv-py2.7.
·· Building ed8136f8 <bm-monkey-patch-is-interactive> for virtualenv-py2.7.
·· Installing ed8136f8 <bm-monkey-patch-is-interactive> into virtualenv-py2.7.
· Profile data does not already exist. Running profiler now.
·· Benchmarking virtualenv-py2.7
··· Running (api.testds.time_create_test_dataset2x2--).
··· api.testds.time_create_test_dataset2x2         
...
```

```shell
$> asv --version
asv 0.4.dev1608+7afea8e5
```

### PR DESCRIPTION
**Pull request by brandonwillard**
Closes #801

### CODE REVIEW

File: asv/results.py [hunk 1]
```diff
@@ -573,7 +573,7 @@ def has_profile(self, benchmark_name):
         """
         Does the given benchmark data have profiling information?
         """
-        return benchmark_name in self._profiles
+        return self._profiles.get(benchmark_name)
```
Comment (brandonwillard) [inline asv/results.py:576]: FYI: `None` entries were present here, which would cause failures when `result.get_profile` was called.

**Review by HaoZeke — APPROVED** [review]
This looks great, could you prefix the commit with `ENH`?

**Comment by HaoZeke** [comment]
Sorry to get back to this so late, could you look into the test failure? I think its ok to also xfail on PyPy with a note

**Comment by brandonwillard** [comment]
> Sorry to get back to this so late, could you look into the test failure? I think its ok to also xfail on PyPy with a note

I can't tell how/if it relates to the changes in this PR, so I can't really help here.

**Comment by HaoZeke** [comment]
Thanks @brandonwillard

**Pull request merged**